docs/tutorials/client-server.md

```diff
@@ -0,0 +1,230 @@
+- Before the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator add a [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown). 
+- Add a [`String`](xref:Bonsai.Expressions.StringProperty) operator in sequence as input to the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer).
+
+:::workflow
+![Basic Dealer input](~/workflows/dealer-basic-input.bonsai)
+:::
+
+- In the operator properties, set the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) `Filter` to the ‘1’ key.
+- Set the [`String`](xref:Bonsai.Expressions.StringProperty) `Value` to ‘Client1’. 
+
+If we run the Bonsai project now, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) will continue listening for incoming messages on the network, but every time the ‘1’ key is pressed a message containing the string ‘Client1’ will be sent from the socket.
+
+- Copy and paste this client structure a couple of times and change the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) and [`String`](xref:Bonsai.Expressions.StringProperty) properties accordingly on each (2, ‘Client2’; 3, ‘Client3’) so that we have 3 total clients that send messages according to different key presses.
+
+:::workflow
+![Multiple clients](~/workflows/multiple-clients.bonsai)
+:::
+
+> [!Note]
+> For the purposes of this article we are creating all of our clients and our server on the same Bonsai project and same machine for ease of demonstration. In a working example, each client and server could be running in separate Bonsai instances on different machines on a network. In this case, localhost would be replaced with the server machine’s IP address.
+
+## Basic server
+Now that we have our client pool set up and sending messages, let’s implement a server to listen for those messages. 
+- Add a [`Router`](xref:Bonsai.ZeroMQ.Router) operator to the project and set its properties to match the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) sockets we already added so that it is running on the same network. 
+- As the [`Router`](xref:Bonsai.ZeroMQ.Router) is acting as server and will be the ‘static’ part of the network, set its `Action` to ‘Bind’.
+
+As with the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator, a [`Router`](xref:Bonsai.ZeroMQ.Router) operator without any input will simply listen for messages on the network and not send anything in return. If we run the project now and monitor the output of the [`Router`](xref:Bonsai.ZeroMQ.Router) operator, we'll see that each time the client sends a message triggered by its associated key press we get a `ResponseContext` produced at the [`Router`](xref:Bonsai.ZeroMQ.Router). Expanding the output the [`Router`](xref:Bonsai.ZeroMQ.Router), we can see it contains a `NetMQMessage`. We [expect](https://netmq.readthedocs.io/en/latest/router-dealer/) this message to be composed of 3 frames: an address (in this case the address of the client that sent the message), an empty delimiter frame and the message content. 
```

Comment:
Reading this now, I wonder whether it might be better to swap the `Router` and the `Dealer` steps. That way you would naturally start with a source operator, rather than a combinator, and you would also have an immediate feedback to test with as soon as you create your first `Dealer`.